### PR TITLE
HW #13 "Multithreading"

### PR DESCRIPTION
**Pull request by sviat-h**

### CODE REVIEW

File: src/com/cursor/hw13/client/Client.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package com.cursor.hw13.client;
+
+import com.cursor.hw13.server.Server;
+
+import java.util.UUID;
+
+public class Client implements Runnable {
+    private Server server = Server.getServer();
+    private String clientID;
```
Comment (YStakhniak) [inline src/com/cursor/hw13/client/Client.java]: clientId

File: src/com/cursor/hw13/client/RequestToServer.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package com.cursor.hw13.client;
+
+public class RequestToServer {
+    private int requestID;
```
Comment (YStakhniak) [inline src/com/cursor/hw13/client/RequestToServer.java]: requestId

File: src/com/cursor/hw13/server/Server.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import com.cursor.hw13.client.RequestToServer;
+
+public class Server {
+    private static Server instance;
+
+    private Server() {
+    }
+
+    public static Server getServer() {
+        if (instance == null) {
```
Comment (YStakhniak) [inline src/com/cursor/hw13/server/Server.java]: The better approach is to use method isNull() from Objects class for null check

File: src/com/cursor/hw13/client/Client.java (hunk 1 above)
Comment (sviat-h) [reply src/com/cursor/hw13/client/Client.java]: done

File: src/com/cursor/hw13/client/RequestToServer.java (hunk 2 above)
Comment (sviat-h) [reply src/com/cursor/hw13/client/RequestToServer.java]: done

File: src/com/cursor/hw13/server/Server.java (hunk 3 above)
Comment (sviat-h) [reply src/com/cursor/hw13/server/Server.java]: done

**Pull request merged**